src/v/cluster/controller_stm.cc

```diff
@@ -62,7 +62,11 @@ controller_stm::maybe_make_snapshot(ssx::semaphore_units apply_mtx_holder) {
     }
 
     controller_snapshot data;
-    // TODO: fill snapshot
+    co_await std::apply(
+      [&data](auto&&... stms) {
+          return ss::when_all(stms.fill_snapshot(data)...);
+      },
+      _state);
```

Comment:
Looking at the interface

```
ss::future<> config_manager::fill_snapshot(controller_snapshot&)
```

it seems like the intention is that `fill_snapshot` will modify `data`. however, since fill_snapshot it is futurized, it seems like there could be a race modifying `data` if `ss::when_all` runs its arguments in parallel?

Reply:
Yeah, the expectation is that sub-stms will fill non-intersecting parts of the snapshot but it is not enforced. Passing some more specific snapshot parts is possible, but it is just so convenient to declare these methods with just a forward declaration of `controller_snapshot`.

Reply:
>fill non-intersecting parts

thanks makes sense.

Reply:
Cwitched to sequential calls just in case. It is not like we gain anything from concurrency, the whole thing is cpu-bound.